### ISSUE TITLE
Inconsistent log format between yamux and other components

### ISSUE THREAD

**Issue by atoom** [hashicorp/consul#8539]
#### Overview of the Issue

Log events written by the `yamux` component is written in another format compared to the rest of the logs from Consul Agent - as a result we get parse failures in our central log management system since the parsing rules assumes that all log lines will look the same.

### Consul info for both Client and Server
```
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 9
        services = 9
build:
        prerelease =
        revision = 8b4a3d95
        version = 1.7.3
consul:
        acl = disabled
        bootstrap = false
        known_datacenters = 1
        leader = true
        leader_addr = 10.248.80.37:8300
        server = true
raft:
        applied_index = 32647466
        commit_index = 32647466
        fsm_pending = 0
        last_contact = 0
        last_log_index = 32647466
        last_log_term = 233
        last_snapshot_index = 32643866
        last_snapshot_term = 233
        latest_configuration = [{Suffrage:Voter ID:43926590-be72-89c2-4e79-ee85ab2cae8a Address:10.248.80.40:8300} {Suffrage:Voter ID:93f2743c-eb92-8053-307f-808c2ec2ff63 Address:10.248.80.37:8300}]
        latest_configuration_index = 0
        num_peers = 1
        protocol_version = 3
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Leader
        term = 233
runtime:
        arch = amd64
        cpu_count = 4
        goroutines = 881
        max_procs = 4
        os = linux
        version = go1.13.7
serf_lan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 94
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 1
        member_time = 243
        members = 10
        query_queue = 0
        query_time = 1
serf_wan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 1
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 1
        member_time = 44
        members = 3
        query_queue = 0
        query_time = 1
```
### Log Fragments

```
Aug 18 10:34:07 node01 consul: 2020-08-18T10:34:07.601+0200 [INFO]  agent.server.raft: compacting logs: from=32205151 to=32221621
Aug 18 10:34:07 node01 consul: 2020-08-18T10:34:07.623+0200 [INFO]  agent.server.raft: snapshot complete up to: index=32231860
Aug 18 10:39:45 node01 consul: 2020-08-18T10:39:39.554+0200 [WARN]  agent.server.raft: heartbeat timeout reached, starting election: last-leader=10.248.80.40:8300
Aug 18 10:39:46 node01 consul: 2020/08/18 10:39:45 [ERR] yamux: Failed to write header: write tcp 10.248.80.37:8300->10.248.80.39:45288: write: broken pipe
Aug 18 10:39:49 node01 consul: 2020-08-18T10:39:46.331+0200 [WARN]  agent: Check is now critical: check=service:example
Aug 18 10:39:49 node01 consul: 2020-08-18T10:39:46.461+0200 [WARN]  agent: Check is now critical: check=service:example
Aug 18 10:39:49 node01 consul: 2020-08-18T10:39:46.785+0200 [WARN]  agent: Check is now critical: check=service:example
Aug 18 10:39:49 node01 consul: 2020-08-18T10:39:47.994+0200 [WARN]  agent: Check is now critical: check=service:example
Aug 18 10:39:49 node01 consul: 2020-08-18T10:39:48.487+0200 [WARN]  agent: Check is now critical: check=service:example
Aug 18 10:39:49 node01 consul: 2020-08-18T10:39:49.202+0200 [WARN]  agent: Check is now critical: check=service:example
Aug 18 10:39:51 node01 consul: 2020/08/18 10:39:51 [ERR] yamux: keepalive failed: i/o deadline reached
Aug 18 10:39:51 node01 consul: 2020/08/18 10:39:51 [ERR] yamux: keepalive failed: i/o deadline reached
Aug 18 10:39:52 node01 consul: 2020/08/18 10:39:52 [ERR] yamux: Failed to write header: write tcp 10.248.80.37:8300->10.248.80.44:54239: write: broken pipe
Aug 18 10:39:53 node01 consul: 2020/08/18 10:39:48 [ERR] yamux: Failed to write header: write tcp 10.248.80.37:43043->10.248.80.40:8300: write: connection reset by peer
Aug 18 10:39:53 node01 consul: 2020/08/18 10:39:53 [ERR] yamux: keepalive failed: write tcp 10.248.80.37:43043->10.248.80.40:8300: write: connection reset by peer
Aug 18 10:39:53 node01 consul: 2020/08/18 10:39:53 [ERR] yamux: Failed to write header: write tcp 10.248.80.37:8300->10.248.80.11:45782: write: broken pipe
Aug 18 10:39:53 node01 consul: 2020/08/18 10:39:52 [ERR] yamux: Failed to write header: write tcp 10.248.80.37:8300->10.248.80.12:41931: write: broken pipe
Aug 18 10:39:54 node01 consul: 2020/08/18 10:39:54 [ERR] yamux: Failed to read stream data: read tcp 10.248.80.37:8300->10.248.80.39:45288: use of closed network connection
Aug 18 10:39:54 node01 consul: 2020/08/18 10:39:54 [ERR] yamux: Failed to write header: write tcp 10.248.80.37:8300->10.248.80.40:35985: write: broken pipe
Aug 18 10:39:54 node01 consul: 2020-08-18T10:39:52.427+0200 [WARN]  agent: Check is now critical: check=service:example
Aug 18 10:39:54 node01 consul: 2020-08-18T10:39:52.758+0200 [WARN]  agent: Check is now critical: check=service:example
Aug 18 10:39:54 node01 consul: 2020-08-18T10:39:52.792+0200 [WARN]  agent: Check is now critical: check=service:example
```

**Comment by dnephin** [comment]
Thank you for reporting this issue!

I believe I fixed this problem as part of #8404. Those changes are merged, but are not yet in a release. They will be released with v1.9.0.  With those changes, all the logs should start with the same prefix format.

**Comment by atoom** [comment]
@dnephin Thanks for the info - I will close this issue for now and assume that it is fixed in 1.9 and if it is not I will re-open it later!
